docs/aws/audit/dynamodbmonitoring/rules/ddb_table_autoscaling_enabled.mdx

```diff
@@ -24,6 +24,285 @@ HIPAA, NIST, AWSWAF
+   ```
+
+   The `read-scaling-policy.json` file should contain the configuration for the target tracking scaling policy, for example:
+
+   ```json
+   {
+       "TargetValue": 70.0,
+       "PredefinedMetricSpecification": {
+           "PredefinedMetricType": "DynamoDBReadCapacityUtilization"
+       },
+       "ScaleOutCooldown": 60,
+       "ScaleInCooldown": 60
+   }
+   ```
+
+4. **Attach a Scaling Policy for Write Capacity:**
+   Similarly, attach a scaling policy for the write capacity.
+
+   ```sh
+   aws application-autoscaling put-scaling-policy \
+       --service-namespace dynamodb \
+       --resource-id table/YourTableName \
+       --scalable-dimension dynamodb:table:WriteCapacityUnits \
+       --policy-name YourWriteScalingPolicyName \
+       --policy-type TargetTrackingScaling \
+       --target-tracking-scaling-policy-configuration file://write-scaling-policy.json
+   ```
+
+   The `write-scaling-policy.json` file should contain the configuration for the target tracking scaling policy, for example:
+
+   ```json
+   {
+       "TargetValue": 70.0,
+       "PredefinedMetricSpecification": {
+           "PredefinedMetricType": "DynamoDBWriteCapacityUtilization"
+       },
+       "ScaleOutCooldown": 60,
+       "ScaleInCooldown": 60
+   }
+   ```
+
+By following these steps, you can ensure that your DynamoDB tables have autoscaling enabled for both read and write capacities using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent DynamoDB tables from being created without autoscaling enabled using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to ensure that autoscaling is enabled for DynamoDB tables:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a DynamoDB Table with Autoscaling**:
+   Use Boto3 to create a DynamoDB table with autoscaling enabled. Below is a sample script to create a table with autoscaling for both read and write capacities:
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon DynamoDB
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   # Initialize DynamoDB and Application Auto Scaling clients
+   dynamodb = session.client('dynamodb')
+   application_autoscaling = session.client('application-autoscaling')
+
+   # Create a DynamoDB table
+   table_name = 'YourTableName'
+   dynamodb.create_table(
+       TableName=table_name,
+       KeySchema=[
+           {
+               'AttributeName': 'PrimaryKey',
+               'KeyType': 'HASH'
+           }
+       ],
+       AttributeDefinitions=[
+           {
+               'AttributeName': 'PrimaryKey',
+               'AttributeType': 'S'
+           }
+       ],
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>